da/dispersal.py

```diff
@@ -0,0 +1,50 @@
+from typing import List, Optional, Generator
+
+from da.common import Certificate, NodeId
+from da.encoder import EncodedData
+from da.verifier import DABlob, Attestation
+
+
+class Dispersal:
+    def __init__(self, nodes: List[NodeId], threshold: int):
+        self.nodes = nodes
+        self.threshold = threshold
+
+    def _prepare_data(self, encoded_data: EncodedData) -> Generator[DABlob, None, None]:
+        assert len(encoded_data.row_commitments) == len(self.nodes)
+        assert len(encoded_data.row_proofs) == len(self.nodes)
+        columns = encoded_data.extended_matrix.columns()
+        column_commitments = encoded_data.column_commitments
+        row_commitments = encoded_data.row_commitments
+        rows_proofs = encoded_data.row_proofs
+        aggregated_column_commitment = encoded_data.aggregated_column_commitment
+        aggregated_column_proof = encoded_data.aggregated_column_proof
+        for index, (column, column_commitment, row_proofs) in enumerate(zip(columns, column_commitments, rows_proofs)):
```

Comment:
nitpick: It seems that we also assert `len(columns) == len(column_commitments) == len(self.nodes)` (or `== len(row_proofs)`, so we can make sure that the `zip` in the line 22 produces the expected number of elements. I came up with this point when reading the `disperse` function below. But, this is not very important. Also, it's okay if the definition of `columns` and `column_commitments` already infer this property.

Reply:
Good catch. I'll add this in subsequent PRs. Thanks!!